README.md

```diff
@@ -427,6 +427,74 @@ You can specify placeholder text for search- and select-type filters by setting
 
 If the filter is search-type, the `placeholder` attribute of the `<input>` element will be set to the placeholder value. If the filter is select-type, the text of the first `<option>` of the `<select>` element will be set to the placeholder value.
 
+#### Custom Filter Components
+
+You may want to render something other than a text box or select list for the filter. To do so, specify the component to render by providing
+the name in the `component` property of the `filter` object.
+
+For example, given that a component named "checkbox-filter" exists:
+
+```javascript
+[
+  {
+    key: 'name',
+    header: 'Name',
+    filter: {
+      component: 'checkbox-filter'
+    }
+  }
+]
+```
+
+Note that the `component` and `type` properties are mutually exclusive.
+
+The components will have access to the `columnDef` and `filter` properites. They contain the column definition and the filter value for
+that column, respectively. Setting the `filter` property in the component will invoke the `onReloadContent` action or cause the manual
+filter buttons to display based on the value of the `realtimeFiltering`.
+
+```handlebars
+{{!-- checkbox-filter.hbs --}}
+<div class="checkbox">
+  <label>
+    {{input type="checkbox" checked=filter}} Check if awesome
+  <label>
+</div>
+```
+
+Debouncing will be applied when the filter is set. If you don't want the delay, set the `debounce` property of the `filter` object to false. 
+
+```javascript
+[
+  {
+    key: 'name',
+    header: 'Name',
+    filter: {
+      component: 'checkbox-filter',
+      debounce: false
```

Comment:
1. I can change this but I chose the default value as true because it is backward-compatible.
2. You're right that they have different configuration options.

The specific workflow I had in mind was a checkbox filter. Since the value can only be true or false, there really isn't a reason to debounce. That said, given your other concerns I don't think that's a strong enough argument to include the option.